---
title: "cbioportal_transcriptomics"
author: "Evan_Naughton"
date: "2024-05-03"
output: html_document
---

```{r}
library(ggplot2)
library(ggpubr)
library(survival)
library(survminer)
library(dplyr)
library(tidyr)
library(reshape2)
library(gridExtra)
library(tidyverse)
library(tidyselect)
library(BiocManager)
library(devtools)
```

```{r}
# Load the sva package
library(sva, lib.loc = "~/R/writable_library")

# Load the immunedeconv package
library(immunedeconv, lib.loc = "~/R/writable_library")
```

```{r}
# Reading in the clinical data

cdata <- read.delim("/home/evannaughton/project/acc_tcga/acc_tcga_clinical_data.tsv", header = T)
head(cdata)
```

```{r}
# Reading in the clinical data from the pancancer atlas data

cdata2 <- read.delim("/home/evannaughton/project/transcriptomic_analysis_cna_quartiles/acc_tcga_pan_can_atlas_2018_clinical_data.tsv", header = T)
head(cdata2)
```

```{r}
# Reading in the genomic data

gdata <- read.delim("/home/evannaughton/project/acc_tcga/data_mrna_seq_v2_rsem.txt", header = T)
head(gdata)
```

```{r}
# Identifying the MKI67 gene
mki67 <- which(gdata$Hugo_Symbol == "MKI67")

# Extracting MKI67 expression values for each sample
mki67_expression <- gdata[mki67, ]  

mki67_expression <- mki67_expression[, -2]

print(mki67_expression)
```

```{r}
# Pivoting to long format

long_df <- pivot_longer(mki67_expression, 
                         cols = -Hugo_Symbol, 
                         names_to = "Sample.ID",  
                         values_to = "Expression_Value")  

long_df <- long_df[, c(2, 1, 3)]

head(long_df)
```

```{r}
# Visualising using a density plot with quantiles

dens <- density(long_df$Expression_Value)
plot(dens, main = "Density Plot of MKI67 Expression Across All 79 Samples", xlab = "MKI67 Expression (RSEM)", ylab = "Density")

quantiles <- quantile(long_df$Expression_Value, probs = c(0.25, 0.5, 0.75))
abline(v = quantiles, col = c("blue", "red", "green"))

legend("topright", legend = c("1st Quartile", "Median", "3rd Quartile"), col = c("blue", "red", "green"), lty = 1)
```

```{r}
# Trying to determine what cutoff points to use for high and low MKI67 expression subgroups.

# Count samples below 1st quartile
samples_below_1st_quartile <- sum(long_df$Expression_Value < quantiles[1])

# Count samples above 3rd quartile
samples_above_3rd_quartile <- sum(long_df$Expression_Value > quantiles[3])

cat("Number of samples below the 1st quartile:", samples_below_1st_quartile, "\n")
cat("Number of samples above the 3rd quartile:", samples_above_3rd_quartile, "\n")
```

```{r}
# Attempting log transformation of the data

long_df <- long_df %>% mutate(Log_Expression_Value = log(Expression_Value))

head(long_df)
```

```{r}
# Visualising log data using a density plot with quantiles

dens2 <- density(long_df$Log_Expression_Value)
plot(dens2, main = "Density Plot of log transformed MKI67 Expression Across All 79 Samples", xlab = "Log MKI67 Expression (RSEM)", ylab = "Density")

quantiles2 <- quantile(long_df$Log_Expression_Value, probs = c(0.25, 0.5, 0.75))
abline(v = quantiles2, col = c("blue", "red", "green"))

legend("topright", legend = c("1st Quartile", "Median", "3rd Quartile"), col = c("blue", "red", "green"), lty = 1)
```

```{r}
# Trying to determine what cutoff points to use for high and low log MKI67 expression subgroups.

# Count samples below 1st quartile
samples_below_1st_quartile2 <- sum(long_df$Log_Expression_Value < quantiles2[1])

# Count samples above 3rd quartile
samples_above_3rd_quartile2 <- sum(long_df$Log_Expression_Value > quantiles2[3])

cat("Number of samples below the 1st quartile:", samples_below_1st_quartile2, "\n")
cat("Number of samples above the 3rd quartile:", samples_above_3rd_quartile2, "\n")
```

There is the same amount of samples in 1st quartile and 3rd quartile even after log transformation.

```{r}
# Quintile plots

dens3 <- density(long_df$Expression_Value)
plot(dens3, main = "Density Plot of MKI67 Expression Across All 79 Samples with quintiles", xlab = "MKI67 Expression (RSEM)", ylab = "Density")

quantiles3 <- quantile(long_df$Expression_Value, probs = seq(0.2, 0.8, by = 0.2))
abline(v = quantiles3, col = c("blue", "red", "green", "purple"), lty = 1)
```

```{r}
# Count samples below 1st quintile
samples_below_1st_quintile <- sum(long_df$Expression_Value < quantiles3[1])

# Count samples above 4th quintile
samples_above_4th_quintile <- sum(long_df$Expression_Value > quantiles3[4])

cat("Number of samples below the 1st quintile:", samples_below_1st_quintile, "\n")
cat("Number of samples above the 4th quintile:", samples_above_4th_quintile, "\n")
```

```{r}
dens4 <- density(long_df$Log_Expression_Value)
plot(dens4, main = "Density Plot of log MKI67 Expression Across All 79 Samples with quintiles", xlab = "Log MKI67 Expression (RSEM)", ylab = "Density")

quantiles4 <- quantile(long_df$Log_Expression_Value, probs = seq(0.2, 0.8, by = 0.2), na.rm = F)
abline(v = quantiles4, col = c("blue", "red", "green", "purple"), lty = 2)
```

```{r}
# Count samples below 1st quintile
samples_below_1st_quintile2 <- sum(long_df$Log_Expression_Value < quantiles4[1])

# Count samples above 4th quintile
samples_above_4th_quintile2 <- sum(long_df$Log_Expression_Value > quantiles4[4])

cat("Number of samples below the 1st quintile:", samples_below_1st_quintile2, "\n")
cat("Number of samples above the 4th quintile:", samples_above_4th_quintile2, "\n")
```

GNOSIS NOT WORKING SO GOING DOWN THIS ROUTE FOR ANALYSIS

```{r}
# Merging dataframes

long_df$Sample.ID <- gsub("\\.", "-", long_df$Sample.ID)

# Merged dataframes based on Sample_ID
finaldf <- merge(cdata, long_df, by = "Sample.ID")
```

```{r}
# Also merging disease free survival data from pancancer atlas dataset

finaldf <- merge(finaldf, cdata2[, c("Sample.ID", "Months.of.disease.specific.survival", "Disease.specific.Survival.status")], by = "Sample.ID", all.x = TRUE)
```

```{r}
# Adding quintile and quartile (and log) columns to the data frame

# Converting 'Expression_Value' column to numeric
finaldf$Expression_Value <- as.numeric(as.character(finaldf$Expression_Value))

# Calculate quartiles and quintiles
quartiles <- quantile(finaldf$Expression_Value, probs = c(0, 0.25, 0.5, 0.75, 1))
quintiles <- quantile(finaldf$Expression_Value, probs = seq(0, 1, by = 0.2))

# Creating quartile column
finaldf$quartile <- cut(finaldf$Expression_Value, breaks = quartiles, labels = FALSE)

# Creating quintile column
finaldf$quintile <- cut(finaldf$Expression_Value, breaks = quintiles, labels = FALSE)
```

```{r}
# calculating log quartiles and log quintiles
log_quartiles <- quantile(finaldf$Log_Expression_Value, probs = c(0, 0.25, 0.5, 0.75, 1))
log_quintiles <- quantile(finaldf$Log_Expression_Value, probs = seq(0, 1, by = 0.2))

# Creating log_quartile column
finaldf$log_quartile <- cut(finaldf$Log_Expression_Value, breaks = log_quartiles, labels = FALSE)

# Creating log_quintile column
finaldf$log_quintile <- cut(finaldf$Log_Expression_Value, breaks = log_quintiles, labels = FALSE)

```

```{r}
# Recoding survival columns

finaldf$OS <- NA

# Setting 'OS' values based on 'Overall.Survival.Status'
finaldf$OS[finaldf$Overall.Survival.Status == "1:DECEASED"] <- 1
finaldf$OS[finaldf$Overall.Survival.Status == "0:LIVING"] <- 0

finaldf$DFS <- NA

# DFS values based on "Disease.Free.Status"
finaldf$DFS[finaldf$Disease.Free.Status == "1:Recurred/Progressed"] <- 1
finaldf$DFS[finaldf$Disease.Free.Status == "0:DiseaseFree"] <- 0

finaldf$DSS <- NA

# DSS values based on "Disease.specific.Survival.status"
finaldf$DSS[finaldf$Disease.specific.Survival.status == "1:DEAD WITH TUMOR"] <- 1
finaldf$DSS[finaldf$Disease.specific.Survival.status == "0:ALIVE OR DEAD TUMOR FREE"] <- 0
```

READY FOR EXPLORATORY ANALYSIS

```{r}
# converting variables to factors to allow for comparisons across groups

finaldf$Sex <- factor(finaldf$Sex)
finaldf$Disease.Free.Status <- factor(finaldf$Disease.Free.Status)
finaldf$Neoplasm.Histologic.Type.Name <- factor(finaldf$Neoplasm.Histologic.Type.Name)
finaldf$Primary.Tumor.Laterality <- factor(finaldf$Primary.Tumor.Laterality)
finaldf$Mitotic.Rate <- factor(finaldf$Mitotic.Rate)
finaldf$Neoplasm.Disease.Stage.American.Joint.Committee.on.Cancer.Code <- factor(finaldf$Neoplasm.Disease.Stage.American.Joint.Committee.on.Cancer.Code)
finaldf$Atypical.Mitotic.Figures <- factor(finaldf$Atypical.Mitotic.Figures)
finaldf$Neoplasm.American.Joint.Committee.on.Cancer.Clinical.Distant.Metastasis.M.Stage <- factor(finaldf$Neoplasm.American.Joint.Committee.on.Cancer.Clinical.Distant.Metastasis.M.Stage)
finaldf$CAPSULAR.INVASION <- factor(finaldf$CAPSULAR.INVASION)
finaldf$Atypical.Mitotic.Figures <- factor(finaldf$Atypical.Mitotic.Figures)
finaldf$Neoplasm.Disease.Lymph.Node.Stage.American.Joint.Committee.on.Cancer.Code <- factor(finaldf$Neoplasm.Disease.Lymph.Node.Stage.American.Joint.Committee.on.Cancer.Code)
finaldf$American.Joint.Committee.on.Cancer.Tumor.Stage.Code <- factor(finaldf$American.Joint.Committee.on.Cancer.Tumor.Stage.Code)

head(finaldf)
```

```{r}
# Creating some boxplots using the log expression values because they are normally distributed

# Sex vs MKI67 expression
sexVmki67 <- ggplot(data = finaldf, aes(x = Sex, y = Log_Expression_Value, color = Sex)) +
  geom_boxplot(alpha = 0, width = 0.5) + stat_summary(fun = median, geom = "text", aes(label = round(..y.., 2)), colour = "black",
               position = position_dodge(width = 0.75), size = 3, vjust = -0.5) +
  stat_compare_means(comparisons = list(c("Male", "Female")), label = "p.format", method = "t.test", size = 5, 
                     vjust = 1.5, label.x = 1.5) +
  geom_jitter(alpha = 0.3, position = position_jitter(width = 0.2)) +
  ggtitle("Sex vs Log MKI67 Expression") +
  ylab("Log MKI67 Expression") +
  xlab("Sex") +
  scale_color_manual(values = c("turquoise", "red")) + 
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18),
    axis.title.x = element_text(hjust = 0.5, size = 18),
    axis.title.y = element_text(hjust = 0.5, size = 18),
    axis.text.x = element_text(size = 15),
    axis.text.y = element_text(size = 15),
    legend.title = element_text(colour = "black", size = 15, face = "bold"),
    legend.text = element_text(size = 15),
    strip.text = element_text(size = 15)
  )

print(sexVmki67)

# Mitotic rate vs Log MKI67 expression

mitoticrateVmki67 <- ggplot(data = finaldf, aes(x = Mitotic.Rate, y = Log_Expression_Value, color = Mitotic.Rate)) +
  geom_boxplot(alpha = 0, width = 0.5) + stat_summary(fun = median, geom = "text", aes(label = round(after_stat(y), 2)), colour = "black",
               position = position_dodge(width = 0.75), size = 3, vjust = -0.5) +
  stat_compare_means(comparisons = list(c("Mitotic Rate > 5/50 HPF Absent", "Mitotic Rate > 5/50 HPF Present")), label = "p.format", method = "t.test", size = 3, 
                     vjust = 1.5, label.x = 1.5) +
  geom_jitter(alpha = 0.3, position = position_jitter(width = 0.2)) +
  ggtitle("Mitotic Rate vs Log MKI67 Expression") +
  ylab("Log MKI67 Expression") +
  xlab("Mitotic Rate") +
  scale_color_manual(values = c("turquoise", "red")) + 
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5, size = 15),
    axis.title.x = element_text(hjust = 0.5, size = 12),
    axis.title.y = element_text(hjust = 0.5, size = 12),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 10),
    legend.title = element_text(colour = "black", size = 10, face = "bold"),
    legend.text = element_text(size = 10),
    strip.text = element_text(size = 10)
  ) 

print(mitoticrateVmki67)

# Capsular Invasion vs Log MKI67 expression

capsularInvasionVmkI67 <- ggplot(data = finaldf, aes(x = CAPSULAR.INVASION, y = Log_Expression_Value, color = CAPSULAR.INVASION)) +
  geom_boxplot(aes(color = CAPSULAR.INVASION), width = 0.5) +
  stat_summary(fun = median, geom = "text", aes(label = round(after_stat(y), 2)), colour = "black", 
               position = position_dodge(width = 0.75), size = 3, vjust = -0.5) +
  stat_compare_means(comparisons = list(c("Invasion of Tumor Capsule Absent", "Invasion of Tumor Capsule Present")), 
                     label = "p.format", method = "t.test", size = 3, 
                     vjust = 1.5, label.x = 1.5) +
  geom_jitter(alpha = 0.3, position = position_jitter(width = 0.2)) +
  ggtitle("Capsular Invasion vs Log MKI67 Expression") +
  ylab("Log MKI67 Expression") +
  scale_color_manual(values = c("turquoise", "red")) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5, size = 15),
    axis.title.x = element_text(hjust = 0.5, size = 12),
    axis.title.y = element_text(hjust = 0.5, size = 12),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 10),
    legend.title = element_text(colour = "black", size = 10, face = "bold"),
    legend.text = element_text(size = 10),
    strip.text = element_text(size = 10)
  )

print(capsularInvasionVmkI67)

# Atypical Mitotic Figures vs Log MKI67 expression

AtypicalMitoticFiguresVmkI67 <- ggplot(data = finaldf, aes(x = Atypical.Mitotic.Figures, y = Log_Expression_Value, color = Atypical.Mitotic.Figures)) +
  geom_boxplot(aes(color = Atypical.Mitotic.Figures), width = 0.5) +
  stat_summary(fun = median, geom = "text", aes(label = round(after_stat(y), 2)), colour = "black", 
               position = position_dodge(width = 0.75), size = 3, vjust = -0.5) +
  stat_compare_means(comparisons = list(c("Atypical Mitotic Figures Absent", "Atypical Mitotic Figures Present")), 
                     label = "p.format", method = "t.test", size = 3, 
                     vjust = 1.5, label.x = 1.5) +
  geom_jitter(alpha = 0.3, position = position_jitter(width = 0.2)) +
  ggtitle("Atypical Mitotic Figures vs Log MKI67 Expression") +
  ylab("Log MKI67 Expression") +
  scale_color_manual(values = c("turquoise", "red")) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5, size = 15),
    axis.title.x = element_text(hjust = 0.5, size = 12),
    axis.title.y = element_text(hjust = 0.5, size = 12),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 10),
    legend.title = element_text(colour = "black", size = 10, face = "bold"),
    legend.text = element_text(size = 10),
    strip.text = element_text(size = 10)
  )

print(AtypicalMitoticFiguresVmkI67)
```

```{r}
# Association scatterplots

# Mitoses count vs LOG MKI67 expression value

scatterplot2 <- ggplot(data = finaldf, aes(x = Mitoses.Count, y = Log_Expression_Value, color = Mitotic.Rate)) +
  geom_point(alpha = 0.5) +  
  geom_smooth(method = "lm", se = FALSE, aes(group = 1), color = "black", linetype = "dashed") +
  ggtitle("Mitoses count vs. Log MKI67 Expression Value with Linear Regression") +
  xlab("Mitoses count") +
  ylab("Log MKI67 Expression Value") +
  theme_bw() +
  scale_color_manual(values = c("blue", "red")) +
  theme(
    plot.title = element_text(hjust = 0.5, size = 12),
    axis.title.x = element_text(hjust = 0.5, size = 12),
    axis.title.y = element_text(hjust = 0.5, size = 12),
    axis.text.x = element_text(size = 10),
    axis.text.y = element_text(size = 10)
  )

print(scatterplot2)

# Mutation count vs LOG MKI67 Expression value
scatterplot3 <- ggplot(data = finaldf, aes(x = Mutation.Count, y = Log_Expression_Value, color = Mitotic.Rate)) +
  geom_point(alpha = 0.5) +  
  geom_smooth(method = "lm", se = FALSE, aes(group = 1), color = "black", linetype = "dashed") + 
  ggtitle("Mutation Count vs. Log MKI67 Expression Value with Linear Regression") +
  xlab("Mutation Count") +
  ylab("Log MKI67 Expression Value") +
  theme_bw() + 
  scale_color_manual(values = c("blue", "red")) +
  theme(
    plot.title = element_text(hjust = 0.5, size = 12),
    axis.title.x = element_text(hjust = 0.5, size = 12),
    axis.title.y = element_text(hjust = 0.5, size = 12),
    axis.text.x = element_text(size = 10),
    axis.text.y = element_text(size = 10)
  )

print(scatterplot3)

# Log MKI67 expression v TMB

scatterp <- ggplot(data = finaldf, aes(x = TMB..nonsynonymous., y = Log_Expression_Value)) +
  geom_point(color = "red", alpha = 0.7) +  
  geom_smooth(method = "lm", se = FALSE, color = "black", linetype = "dashed") +
  ggtitle("TMB (Nonsynonymous) vs. Log MKI67 Expression Value with Linear Regression") +
  xlab("Tumour Mutational Burden") +
  ylab("Log MKI67 Expression Value") +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5, size = 12),
    axis.title.x = element_text(hjust = 0.5, size = 12),
    axis.title.y = element_text(hjust = 0.5, size = 12),
    axis.text.x = element_text(size = 10),
    axis.text.y = element_text(size = 10)
  )

print(scatterp)

# Log MKI67 expression v fraction genome altered

scatterp1 <- ggplot(data = finaldf, aes(x = Fraction.Genome.Altered, y = Log_Expression_Value)) +
  geom_point(color = "red", alpha = 0.7) +  
  geom_smooth(method = "lm", se = FALSE, color = "black", linetype = "dashed") +
  ggtitle("Fraction genome altered vs. Log MKI67 Expression Value with Linear Regression") +
  xlab("Fraction Genome Altered") +
  ylab("Log MKI67 Expression Value") +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5, size = 12),
    axis.title.x = element_text(hjust = 0.5, size = 12),
    axis.title.y = element_text(hjust = 0.5, size = 12),
    axis.text.x = element_text(size = 10),
    axis.text.y = element_text(size = 10)
  )

print(scatterp1)

lmfit <- lm(Log_Expression_Value ~ TMB..nonsynonymous., data = finaldf)
summary(lmfit)

lmfit2 <- lm(Log_Expression_Value ~ Mitoses.Count, data = finaldf)
summary(lmfit2)

lmfit3 <- lm(Log_Expression_Value ~ Mutation.Count, data = finaldf)
summary(lmfit3)

lmfit4 <- lm(Log_Expression_Value ~ Fraction.Genome.Altered, data = finaldf)
summary(lmfit4)
```

```{r}
# SURVIVAL ANALYSIS 

# survival estimates for OS
surv_os <- survfit(Surv(Overall.Survival..Months., OS) ~ 1, data = finaldf)

# survival estimates for DFS
surv_dfs <- survfit(Surv(Disease.Free..Months., DFS) ~ 1, data = finaldf)

# Plotting overall survival curve
ggsurvplot(surv_os, data = finaldf, risk.table = TRUE, pval = TRUE,
           conf.int = TRUE, legend.title = "Strata",
           surv.median.line = "hv", 
           title = "Overall Survival",
           xlab = "Time (months)", ylab = "Survival Probability")

# Plotting DFS survival curve
ggsurvplot(surv_dfs, data = finaldf, risk.table = TRUE, pval = TRUE,
           conf.int = TRUE, legend.title = "Strata",
           surv.median.line = "hv", 
           title = "Disease Free Survival",
           xlab = "Time (months)", ylab = "Survival Probability")
```

```{r}
# Survival Analysis based on MKI67 expression quartiles

surv_models <- survfit(Surv(Overall.Survival..Months., OS) ~ quartile, data = finaldf)
surv_models2 <- survfit(Surv(Disease.Free..Months., DFS) ~ quartile, data = finaldf)
surv_models3 <- survfit(Surv(Months.of.disease.specific.survival, DSS) ~ quartile, data = finaldf)

# Plot survival curves for each quartile group with risk table
ggsurvplot(surv_models, data = finaldf, pval = TRUE,
           legend.title = "Quartile",
           title = "OS Curves by Quartile of MKI67 Expression",
           xlab = "Time (months)", ylab = "Survival Probability",
           risk.table = TRUE, 
           risk.table.title = "Risk Table", 
           risk.table.col = "strata", 
           tables.height = 0.4, 
           theme = theme(plot.margin = margin(t = 2, r = 2, b = 2, l = 2, unit = "cm")),
           size = 0.8)

ggsurvplot(surv_models2, data = finaldf, pval = TRUE,
           legend.title = "Quartile",
           title = "DFS Curves by Quartile of MKI67 Expression",
           xlab = "Time (months)", ylab = "Survival Probability",
           risk.table = TRUE, 
           risk.table.title = "Risk Table", 
           risk.table.col = "strata", 
           tables.height = 0.4, 
           theme = theme(plot.margin = margin(t = 2, r = 2, b = 2, l = 2, unit = "cm")),
           size = 0.8)

ggsurvplot(surv_models3, data = finaldf, pval = TRUE,
           legend.title = "Quartile",
           title = "DSS Curves by Quartile of MKI67 Expression",
           xlab = "Time (months)", ylab = "Survival Probability",
           risk.table = TRUE, 
           risk.table.title = "Risk Table", 
           risk.table.col = "strata", 
           tables.height = 0.4, 
           theme = theme(plot.margin = margin(t = 2, r = 2, b = 2, l = 2, unit = "cm")),
           size = 0.8)

# Perform log-rank test for quartile groups
quartile_logrank_test <- survdiff(Surv(Overall.Survival..Months., OS) ~ quartile, data = finaldf)
quartile_logrank_test

quartile_logrank_test2 <- survdiff(Surv(Disease.Free..Months., DFS) ~ quartile, data = finaldf)
quartile_logrank_test2

quartile_logrank_test3 <- survdiff(Surv(Months.of.disease.specific.survival, DSS) ~ quartile, data = finaldf)
quartile_logrank_test3
```

```{r}
# Fitting a survival model for each quintile group
surv_models_quintiles <- survfit(Surv(Overall.Survival..Months., OS) ~ quintile, data = finaldf)

# Plot survival curves for each quintile group with risk table
ggsurvplot(surv_models_quintiles, data = finaldf, pval = TRUE,
           conf.int = TRUE, legend.title = "Quintile",
           title = "Survival Curves by Quintile of MKI67 Expression",
           xlab = "Time (months)", ylab = "Survival Probability",
           risk.table = TRUE, 
           risk.table.title = "Risk Table", 
           risk.table.col = "strata", 
           tables.height = 0.4, 
           theme = theme(plot.margin = margin(t = 2, r = 2, b = 2, l = 2, unit = "cm")),
           size = 0.8) 

# Perform log-rank test for quintile groups
quintile_logrank_test <- survdiff(Surv(Overall.Survival..Months., OS) ~ quintile, data = finaldf)
quintile_logrank_test
```

```{r}
# Filtering the data for patients in quartiles 1 and 4
quartile_data <- finaldf %>% 
  filter(quartile %in% c(1, 4))

# Fit survival model for quartiles 1 and 4 combined
surv_model2 <- survfit(Surv(Overall.Survival..Months., OS) ~ quartile, data = quartile_data)

# Plot Kaplan-Meier curves for quartiles 1 and 4
ggsurvplot(surv_model2, data = quartile_data, pval = TRUE, palette = c("turquoise", "red"),
           legend.labs = c("Quartile 1", "Quartile 4"),
           conf.int = TRUE,
           title = "Survival Curves for MKI67 Expression Quartiles 1 and 4",
           xlab = "Time (months)", ylab = "Survival Probability", risk.table = TRUE, 
           risk.table.title = "Risk Table", 
           risk.table.col = "strata", 
           tables.height = 0.3, 
           theme = theme(plot.margin = margin(t = 2, r = 2, b = 2, l = 2, unit = "cm")),
           size = 0.8) 

# Log-Rank test
quartile_1_4_logrank_test <- survdiff(Surv(Overall.Survival..Months., OS) ~ quartile, data = quartile_data)
quartile_1_4_logrank_test
```
```{r}
# Fitting Disease free survival for quartiles 1 and 4 combined
surv_model3 <- survfit(Surv(Disease.Free..Months., DFS) ~ quartile, data = quartile_data)

# Plot Kaplan-Meier curves for quartiles 1 and 4
ggsurvplot(surv_model3, data = quartile_data, pval = TRUE, palette = c("turquoise", "red"),
           legend.labs = c("Quartile 1", "Quartile 4"),
           conf.int = TRUE,
           title = "Disease Free Survival Curves for MKI67 Expression Quartiles 1 and 4",
           xlab = "Time (months)", ylab = "Survival Probability", risk.table = TRUE, 
           risk.table.title = "Risk Table", 
           risk.table.col = "strata", 
           tables.height = 0.3, 
           theme = theme(plot.margin = margin(t = 2, r = 2, b = 2, l = 2, unit = "cm")),
           size = 0.8) 

# Log-Rank test
quartile_1_4_logrank_test2 <- survdiff(Surv(Disease.Free..Months., DFS) ~ quartile, data = quartile_data)
quartile_1_4_logrank_test2
```

```{r}
# Fitting Disease specific survival for quartiles 1 and 4 combined
surv_model4 <- survfit(Surv(Months.of.disease.specific.survival, DSS) ~ quartile, data = quartile_data)

# Plot Kaplan-Meier curves for quartiles 1 and 4
ggsurvplot(surv_model4, data = quartile_data, pval = TRUE, palette = c("turquoise", "red"),
           legend.labs = c("Quartile 1", "Quartile 4"),
           conf.int = TRUE,
           title = "Disease Specific Survival Curves for MKI67 Expression Quartiles 1 and 4",
           xlab = "Time (months)", ylab = "Survival Probability", risk.table = TRUE, 
           risk.table.title = "Risk Table", 
           risk.table.col = "strata", 
           tables.height = 0.3, 
           theme = theme(plot.margin = margin(t = 2, r = 2, b = 2, l = 2, unit = "cm")),
           size = 0.8) 

# Log-Rank test
quartile_1_4_logrank_test3 <- survdiff(Surv(Months.of.disease.specific.survival, DSS) ~ quartile, data = quartile_data)
quartile_1_4_logrank_test3
```

```{r}
# Fitting a Multivariable Cox Proportional Hazards Model to the DSS survival plot to identify potential confounding variables
quartile_data$quartile <- factor(quartile_data$quartile)

# Create the survival object
surv_cox <- Surv(quartile_data$Months.of.disease.specific.survival, quartile_data$DSS)

# Fit the Cox Proportional Hazards Model
cox_model <- coxph(surv_cox ~ quartile + Diagnosis.Age + Mitotane.Therapy + Sex, data = quartile_data)

# Summarize the model
summary(cox_model)
```

The hazard for quartile 4 is 16.909 times higher than for quartile 1, and this difference is statistically significant (p < 0.001).
The age at diagnosis does not significantly affect the hazard, as indicated by a p-value of 0.969.
The hazard ratio for patients undergoing Mitotane therapy is 3.149, suggesting an increased risk, but this is not statistically significant at the 0.05 level (p ≈ 0.083).
Being male is associated with a hazard ratio of 2.163 compared to females, but this result is not statistically significant (p ≈ 0.199).

  Concordance: 0.867 (se = 0.031)
      A concordance index of 0.867 indicates good predictive accuracy.
      
  Likelihood ratio test: 29.61 on 4 df, p = 6e-06
      The model is statistically significant overall.
      
  Wald test: 17.62 on 4 df, p = 0.001
      The predictors are jointly significant.
      
  Score (logrank) test: 28.98 on 4 df, p = 8e-06
      The model is statistically significant.
      
The primary variable of interest, quartile4, shows a significant and large effect on survival, indicating that being in quartile 4 significantly increases the hazard compared to quartile 1.
Other variables (Diagnosis.Age, Mitotane.Therapy, and Sex) did not show statistically significant effects on the hazard in this model.
The model as a whole fits well, as indicated by the concordance index and the significance of the overall tests. 

Creating a Forest Plot:

```{r}
# Forest plot for overall survival 
model <- coxph(Surv(quartile_data$Months.of.disease.specific.survival, quartile_data$DSS) ~ quartile + Diagnosis.Age + Sex + Mitotane.Therapy + Primary.Tumor.Laterality + Surgical.Margin.Resection.Status, data = quartile_data)

ggforest(
  model,
  data = NULL,
  main = "Hazard ratio (DSS)",
  cpositions = c(0.02, 0.22, 0.4),
  fontsize = 0.7,
  refLabel = "reference",
  noDigits = 2
)
```

```{r}
# Forest plot for DSS

```


```{r}
# Plotting the data using cutpoints
cutpoint2 <- surv_cutpoint(
  data = finaldf,
  time = "Months.of.disease.specific.survival",
  event = "DSS",
  variables = "Expression_Value"
)

# Plot the cutpoint
plot(cutpoint2)

# Extract the cutpoint value
cutpoint_value2 <- cutpoint2$cutpoint$cutpoint

# Use the cutpoint to create a new column indicating the group
finaldf$group <- ifelse(finaldf$Expression_Value > cutpoint_value2, "high", "low")

# Plot the survival curves based on the new group variable
fit2 <- survfit(Surv(Months.of.disease.specific.survival, DSS) ~ group, data = finaldf)

ggsurvplot(fit2, data = finaldf, pval = TRUE, palette = c("red", "turquoise"),
           legend.labs = c("High MKI67", "Low MKI67"),
           title = "Cutpoint Survival - High & Low MKI67 Expression",
           xlab = "Time (months)", ylab = "Survival Probability", risk.table = TRUE, 
           risk.table.title = "Risk Table", 
           risk.table.col = "strata", 
           tables.height = 0.3, 
           theme = theme(plot.margin = margin(t = 2, r = 2, b = 2, l = 2, unit = "cm")),
           size = 0.8) 

# Log-Rank test
quartile_1_4_logrank_test4 <- survdiff(Surv(Months.of.disease.specific.survival, DSS) ~ group, data = finaldf)
quartile_1_4_logrank_test4
```

```{r}
#write.table(finaldf, file = "finaldf.txt", sep = "\t", row.names = FALSE)
```

```{r}
# Perform Kruskal-Wallis test
kruskal_result <- kruskal.test(Expression_Value ~ quartile, data = finaldf)

# Summarize Kruskal-Wallis test results
print(kruskal_result)

# Perform pairwise Wilcoxon tests
pairwise_result <- pairwise.wilcox.test(finaldf$Expression_Value, finaldf$quartile, p.adjust.method = "bonferroni")

# Summarize pairwise Wilcoxon test results
print(pairwise_result)

```

```{r}
# Violin plot to visualize the distribution of MKI67 expression values across tumor stages
finaldf2 <- finaldf[!is.na(finaldf$Neoplasm.Disease.Stage.American.Joint.Committee.on.Cancer.Code), ]
ggplot(data = finaldf2, aes(x = Neoplasm.Disease.Stage.American.Joint.Committee.on.Cancer.Code, y = Expression_Value, fill = Neoplasm.Disease.Stage.American.Joint.Committee.on.Cancer.Code)) +
  geom_violin(trim = FALSE) +
  geom_boxplot(width = 0.1, fill = "white", color = "black") +  
  labs(x = "Disease Stage", y = "MKI67 Expression") +
  theme_minimal() +
  guides(fill = guide_legend(title = NULL))
```

```{r}
# Perform Kruskal-Wallis test
kruskal_result <- kruskal.test(Expression_Value ~ Neoplasm.Disease.Stage.American.Joint.Committee.on.Cancer.Code, data = finaldf)

# Summarize Kruskal-Wallis test results
print(kruskal_result)

# Perform pairwise Wilcoxon tests
pairwise_result <- pairwise.wilcox.test(finaldf$Expression_Value, finaldf$Neoplasm.Disease.Stage.American.Joint.Committee.on.Cancer.Code, p.adjust.method = "bonferroni")

# Summarize pairwise Wilcoxon test results
print(pairwise_result)
```

```{r}
# Violin plot to visualize the distribution of Log MKI67 expression values across tumor stages
finaldf2 <- finaldf[!is.na(finaldf$Neoplasm.Disease.Stage.American.Joint.Committee.on.Cancer.Code), ]
ggplot(data = finaldf2, aes(x = Neoplasm.Disease.Stage.American.Joint.Committee.on.Cancer.Code, y = Log_Expression_Value, fill = Neoplasm.Disease.Stage.American.Joint.Committee.on.Cancer.Code)) +
  geom_violin(trim = FALSE) +
  geom_boxplot(width = 0.1, fill = "white", color = "black") +  
  labs(x = "Disease Stage", y = "Log MKI67 Expression") +
  theme_minimal() +
  guides(fill = guide_legend(title = NULL))
```

```{r}
# Perform ANOVA
anova_result <- aov(Log_Expression_Value ~ Neoplasm.Disease.Stage.American.Joint.Committee.on.Cancer.Code, data = finaldf)

# Summarize ANOVA results
summary(anova_result)
```

FROM THE RESULTS, MKI67 EXPRESSION IS SIGNIFICANTLY ASSOCIATED WITH TUMOUR STAGE

Looking for correlations between MKI67 expression and other tumour characteristics:

```{r}
# Violin plot to visualize the distribution of Log MKI67 expression values across tumor stage code
finaldf2 <- finaldf[!is.na(finaldf$American.Joint.Committee.on.Cancer.Tumor.Stage.Code), ]
ggplot(data = finaldf2, aes(x = American.Joint.Committee.on.Cancer.Tumor.Stage.Code, y = Log_Expression_Value, fill = American.Joint.Committee.on.Cancer.Tumor.Stage.Code)) +
  geom_violin(trim = FALSE) +
  geom_boxplot(width = 0.1, fill = "white", color = "black") +  
  labs(x = "Tumour Stage", y = "Log MKI67 Expression") +
  theme_minimal() +
  guides(fill = guide_legend(title = NULL))
```

```{r}
# Perform ANOVA
anova_result2 <- aov(Log_Expression_Value ~ American.Joint.Committee.on.Cancer.Tumor.Stage.Code, data = finaldf)

# Summarize ANOVA results
summary(anova_result2)
```

```{r}
# Violin plot to visualize the distribution of Log MKI67 expression values across Lymph Node stage
finaldf2 <- finaldf[!is.na(finaldf$Neoplasm.Disease.Lymph.Node.Stage.American.Joint.Committee.on.Cancer.Code), ]
ggplot(data = finaldf2, aes(x = Neoplasm.Disease.Lymph.Node.Stage.American.Joint.Committee.on.Cancer.Code, y = Log_Expression_Value, fill = Neoplasm.Disease.Lymph.Node.Stage.American.Joint.Committee.on.Cancer.Code)) +
  geom_violin(trim = FALSE) +
  geom_boxplot(width = 0.1, fill = "white", color = "black") +  
  labs(x = "Lymph Node Stage", y = "Log MKI67 Expression") +
  theme_minimal() +
  guides(fill = guide_legend(title = NULL))
```

```{r}
# Perform t-test
t_test_result <- t.test(Log_Expression_Value ~ Neoplasm.Disease.Lymph.Node.Stage.American.Joint.Committee.on.Cancer.Code, data = finaldf)

print(t_test_result)
```

```{r}
# Violin plot to visualize the distribution of Log MKI67 expression values across metastasis stage
finaldf2 <- finaldf[!is.na(finaldf$Neoplasm.American.Joint.Committee.on.Cancer.Clinical.Distant.Metastasis.M.Stage), ]
ggplot(data = finaldf2, aes(x = Neoplasm.American.Joint.Committee.on.Cancer.Clinical.Distant.Metastasis.M.Stage, y = Log_Expression_Value, fill = Neoplasm.American.Joint.Committee.on.Cancer.Clinical.Distant.Metastasis.M.Stage)) +
  geom_violin(trim = FALSE) +
  geom_boxplot(width = 0.1, fill = "white", color = "black") +  
  labs(x = "Metastasis Stage", y = "Log MKI67 Expression") +
  theme_minimal() +
  guides(fill = guide_legend(title = NULL))
```

```{r}
# Perform t-test
t_test_result <- t.test(Log_Expression_Value ~ Neoplasm.American.Joint.Committee.on.Cancer.Clinical.Distant.Metastasis.M.Stage, data = finaldf)

print(t_test_result)
```



xCell analysis: 

```{r}
#devtools::install_github('dviraran/xCell')
library(xCell)
```

```{r}
# reading in data
xcell <- read.table("/home/evannaughton/project/r_code/countdata_highlow.txt", header = T)
xcell <- subset(xcell, select = -Entrez_Gene_Id)
xcell <- subset(xcell, select = -row.names)
rownames(xcell) = make.names(xcell$Hugo_Symbol, unique = TRUE)
xcell <- subset(xcell, select = -Hugo_Symbol)
```

```{r}
xcell = as.matrix(xcell)
class(xcell)
```

```{r}
immune_cells = c("B-cells", "Basophils", "CD4+ memory T-cells", "CD4+ T-cells", "CD4+ Tcm", "CD4+ Tem", "CD8+ T-cells", "CD8+ Tcm", "CD8+ Tem", "Class-switched memory B-cells", "Eosinophils", "Macrophages", "Macrophages M1", "Macrophages M2", "Mast cells", "Memory B-cells", "Monocytes", "Neutrophils", "NK cells", "NKT", "Plasma cells", "pro B-cells", "Th1 cells", "Th2 cells", "Tregs", "ImmuneScore", "StromaScore", "MicroenvironmentScore")

immune_cells1 = c("B-cells", "Basophils", "CD4+ memory T-cells", "CD4+ T-cells", "CD4+ Tcm", "CD4+ Tem", "CD8+ T-cells", "CD8+ Tcm", "CD8+ Tem", "Class-switched memory B-cells", "Eosinophils", "Macrophages", "Macrophages M1", "Macrophages M2")

immune_cells2 = c("Mast cells", "Memory B-cells", "Monocytes", "Neutrophils", "NK cells", "NKT", "Plasma cells", "pro B-cells", "Th1 cells", "Th2 cells", "Tregs", "ImmuneScore", "StromaScore", "MicroenvironmentScore")
```

```{r}
# xCell analysis

xcell_data <- xCellAnalysis(xcell, rnaseq = T)
```

```{r}
# Reading in genomic quartile data

gdata_quartile1 <- read.table("/home/evannaughton/project/r_code/gdata_quartile1", header = T)
gdata_quartile4 <- read.table("/home/evannaughton/project/r_code/gdata_quartile4", header = T)

# Subset the xCell results matrix for high expression samples
high_expression_matrix <- xcell_data[, colnames(xcell_data) %in% colnames(gdata_quartile4)]

# Subset the xCell results matrix for low expression samples
low_expression_matrix <- xcell_data[, colnames(xcell_data) %in% colnames(gdata_quartile1)]

# Ensure row names are retained
row.names(high_expression_matrix) <- row.names(xcell_data)
row.names(low_expression_matrix) <- row.names(xcell_data)
```


```{r}
# Remove special characters from row names
#rownames(low_expression_matrix) <- gsub("[^A-Za-z0-9_]+", "_", rownames(low_expression_matrix))
#rownames(high_expression_matrix) <- gsub("[^A-Za-z0-9_]+", "_", rownames(high_expression_matrix))
```

```{r}
# reading in sample IDs

high_sample_ids <- read.table("/home/evannaughton/project/r_code/high_sample_ids", header = T)
low_sample_ids <- read.table("/home/evannaughton/project/r_code/low_sample_ids", header = T)
high_sample_ids <- high_sample_ids %>% pull(x)
low_sample_ids <- low_sample_ids %>% pull(x)

# Filter xCell results based on sample IDs
xcell_results_high1 <- xcell_data[immune_cells1, colnames(xcell_data) %in% high_sample_ids]
xcell_results_low1 <- xcell_data[immune_cells1, colnames(xcell_data) %in% low_sample_ids]

# Melt data for creating boxplots
melted_high_group1 <- melt(xcell_results_high1)
melted_low_group1 <- melt(xcell_results_low1)

# Boxplots for high expression group 
high_plot1 <- ggplot(melted_high_group1, aes(x=Var1, y=value)) +
  geom_boxplot() +
  labs(title="Cell Abundances in High Expression Group",
       x="Cell Type", y="Cell Abundance") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +  
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Boxplots for low expression group 
low_plot1 <- ggplot(melted_low_group1, aes(x=Var1, y=value)) +
  geom_boxplot() +
  labs(title="Cell Abundances in Low Expression Group",
       x="Cell Type", y="Cell Abundance") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

high_plot1
low_plot1
```

```{r}
# Merge plots
merged_plot <- grid.arrange(high_plot1, low_plot1, ncol=2)

# Print merged plot
print(merged_plot)
```

```{r}
# Combine filtered data for both groups into a single data frame
merged_data1 <- rbind(transform(melted_high_group1, group = "High Expression"),
                     transform(melted_low_group1, group = "Low Expression"))
```

```{r}
# Initialize vector to store p-values
p_values1 <- numeric(length = length(unique(merged_data1$Var1)))

# Perform t-test between low and high expression groups for each cell type
for (i in seq_along(p_values1)) {
  cell_type <- unique(merged_data1$Var1)[i]
  subset_data <- merged_data1[merged_data1$Var1 == cell_type, ]
  t_test_result <- t.test(value ~ group, data = subset_data)
  p_values1[i] <- t_test_result$p.value
}
```

```{r}
# Get maximum value of 'value' variable for each cell type
max_values1 <- aggregate(value ~ Var1, merged_data1, max)

# grouped boxplot with p-values
grouped_boxplot1 <- ggplot(merged_data1, aes(x = Var1, y = value, fill = group)) +
  geom_boxplot(position = position_dodge(width = 0.8), outlier.size = 0.5) +  
  labs(title = "Tumour-Infiltrating Immune Cell Abundances - High & Low MKI67 Expression",
       x = NULL, y = "Cell Abundance") +
   scale_fill_manual(values = c("High Expression" = "red", "Low Expression" = "turquoise")) +
  theme_minimal() +
   theme(axis.text.x = element_text(angle = 45, hjust = 1)) 

# Add p-values to the plot with adjusted heights for specific cell types
for (i in seq_along(p_values1)) {
  cell_type <- unique(merged_data1$Var1)[i]
  subset_data <- merged_data1[merged_data1$Var1 == cell_type, ]
  max_y <- max(subset_data$value, na.rm = TRUE)
  p_value <- p_values1[i]
  
  # Adjust height for specific cell types
  if (cell_type %in% c("Eosinophils", "CD8+ Tem", "CD4+ memory T-cells", "CD4+ T-cells", "Macrophages", "Macrophages M1")) {
    p_value_height <- max_y * 1.5  # Increase height for specified cell types
  } else {
    p_value_height <- max_y * 1.05  # Default height for other cell types
  }
  
  grouped_boxplot1 <- grouped_boxplot1 +
    annotate("text", x = i, y = p_value_height, label = sprintf("p = %.4f", p_value), size = 2, color = "black")
}

# Print grouped boxplot
print(grouped_boxplot1)
```

```{r}
# Boxplots 2

# Filter xCell results based on sample IDs
xcell_results_high2 <- xcell_data[immune_cells2, colnames(xcell_data) %in% high_sample_ids]
xcell_results_low2 <- xcell_data[immune_cells2, colnames(xcell_data) %in% low_sample_ids]

# Melt the data for creating boxplots
melted_high_group2 <- melt(xcell_results_high2)
melted_low_group2 <- melt(xcell_results_low2)

# Combine filtered data for both groups into a single data frame
merged_data2 <- rbind(transform(melted_high_group2, group = "High Expression"),
                     transform(melted_low_group2, group = "Low Expression"))
```

```{r}
# Initialize vector to store p-values
p_values2 <- numeric(length = length(unique(merged_data2$Var1)))

# Perform t-test between low and high expression groups for each cell type
for (i in seq_along(p_values2)) {
  cell_type <- unique(merged_data2$Var1)[i]
  subset_data <- merged_data2[merged_data2$Var1 == cell_type, ]
  t_test_result <- t.test(value ~ group, data = subset_data)
  p_values2[i] <- t_test_result$p.value
}
```

```{r}
# Get maximum value of 'value' variable for each cell type
max_values2 <- aggregate(value ~ Var1, merged_data2, max)

# Create grouped boxplot with p-values
grouped_boxplot2 <- ggplot(merged_data2, aes(x = Var1, y = value, fill = group)) +
  geom_boxplot(position = position_dodge(width = 0.8), outlier.size = 0.5) +  
  labs(title = "Tumour-Infiltrating Immune Cell Abundances - High & Low MKI67 Expression",
       x = NULL, y = "Cell Abundance") +
   scale_fill_manual(values = c("High Expression" = "red", "Low Expression" = "turquoise")) +
  theme_minimal() +
   theme(axis.text.x = element_text(angle = 45, hjust = 1)) 

# Add p-values to the plot
for (i in seq_along(p_values2)) {
  cell_type <- unique(merged_data2$Var1)[i]
  subset_data <- merged_data2[merged_data2$Var1 == cell_type, ]
  max_y <- max(subset_data$value, na.rm = TRUE)
  p_value <- p_values2[i]
  
# Adjust height for specific cell types
  if (cell_type %in% c("Mast cells", "Memory B-cells", "Neutrophils", "NK cells", "Plasma cells")) {
    p_value_height <- max_y * 2  
  } else {
    p_value_height <- max_y * 1.1  
  }
  
  grouped_boxplot2 <- grouped_boxplot2 +
    annotate("text", x = i, y = p_value_height, label = sprintf("p = %.4f", p_value), size = 2, color = "black")
}

# Print grouped boxplot
print(grouped_boxplot2)
```

```{r}
# Printing t-test results for all immune cell types

# Filter xCell results based on sample IDs

xcell_results_high <- xcell_data[immune_cells, colnames(xcell_data) %in% high_sample_ids]
xcell_results_low <- xcell_data[immune_cells, colnames(xcell_data) %in% low_sample_ids]

# Melt the data for creating boxplots

melted_high_group <- melt(xcell_results_high)
melted_low_group <- melt(xcell_results_low)

# Combine filtered data for both groups into a single data frame

merged_data <- rbind(transform(melted_high_group, group = "High Expression"),
                     transform(melted_low_group, group = "Low Expression"))

# Perform t-test for each immune cell type

for (cell_type in unique(merged_data$Var1)) {
  subset_data <- merged_data[merged_data$Var1 == cell_type, ]
  if (all(c("High Expression", "Low Expression") %in% subset_data$group)) {
    t_test_result <- t.test(value ~ group, data = subset_data)
    cat("T-test results for", cell_type, ":\n")
    print(t_test_result)
    cat("\n")
  } else {
    cat("Insufficient data for", cell_type, "\n\n")
  }
}
```

CIBERSORT ANALYSIS:

```{r}
cibersort <- read.delim("/home/evannaughton/project/CIBERSORT_output_files/CIBERSORTx_Job5_output/CIBERSORTxGEP_Job5_Fractions-Adjusted.txt", header = TRUE, row.names = 1)
head(cibersort)
```

```{r}
# Adding a group column to the CIBERSORT results
cibersort <- cibersort %>%
  rownames_to_column("Sample") %>%
  mutate(group = case_when(
    Sample %in% quartile1_ids ~ "Low Expression",
    Sample %in% quartile4_ids ~ "High Expression",
    TRUE ~ NA_character_  
  ))
```

```{r}
# Melt the data
cibersort_long <- cibersort %>%
  pivot_longer(cols = -c(Sample, group), names_to = "Cell_Type", values_to = "Abundance")

# Ensure cibersort_long is a data frame
cibersort_long <- as.data.frame(cibersort_long)

# Remove specific rows from the cibersort_long dataframe
cibersort_long <- cibersort_long %>%
  filter(!Cell_Type %in% c("P.value", "RMSE", "Correlation"))

# Check the structure of the updated dataframe
str(cibersort_long)

# Set options to display numbers in fixed decimal format
options(scipen = 999, digits = 6)

# Create the boxplot without log transformation
p <- ggplot(cibersort_long, aes(x = Cell_Type, y = Abundance, fill = group)) +
  geom_boxplot(outlier.size = 0.5) +  
  labs(title = "    Tumour-Infiltrating Immune Cell Abundances - High & Low MKI67 Expression",
       y = "Cell Abundance", x = "") +
  scale_fill_manual(values = c("High Expression" = "red", "Low Expression" = "turquoise")) +
  scale_y_continuous(breaks = seq(0, 0.375, 0.05), limits = c(0, 0.375)) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    plot.title = element_text(hjust = 0.5)
  )

print(p)
```

```{r}
# Filter out cell types with constant values
filtered_cibersort_long <- cibersort_long %>%
  group_by(Cell_Type) %>%
  filter(sd(Abundance) > 0)

# Perform t-tests for each cell type
t_test_results <- filtered_cibersort_long %>%
  group_by(Cell_Type) %>%
  summarise(
    t_test = list(t.test(Abundance ~ group)),
    p_value = map_dbl(t_test, ~ .x$p.value),
    mean_diff = map_dbl(t_test, ~ diff(.x$estimate))
  )

print(t_test_results)
```

Using immunedeconv for immune cell analysis (MCP-counter):

```{r}
# creating expression matrix
gdata_high_low <- gdata_high_low[, !(colnames(gdata_high_low) %in% c("Hugo_Symbol", "Entrez_Gene_Id"))]
gdata_high_low <- as.matrix(gdata_high_low)
```

```{r}
# Create a vector of conditions corresponding to the samples
conditions <- rep(NA, ncol(gdata_high_low))
conditions[colnames(gdata_high_low) %in% quartile1_ids] <- "low"
conditions[colnames(gdata_high_low) %in% quartile4_ids] <- "high"
```

```{r}
# Perform deconvolution using the MCP-counter method
results <- deconvolute(gdata_high_low, method = "mcp_counter")
```

```{r}
# Convert the results to a data frame
results_df <- as.data.frame(t(results))

# Reassign the first row as column names
colnames(results_df) <- as.character(results_df[1, ])
results_df <- results_df[-1, ]

# Ensure that all numeric columns are properly converted to numeric type
results_df <- results_df %>%
  mutate(across(where(is.character), as.numeric))

# Now add the conditions information
results_df$group <- conditions

# Proceed with creating the long format for plotting
results_long <- melt(results_df, id.vars = "group", variable.name = "Cell_Type", value.name = "Abundance")
```

```{r}
# Ensure correct factor levels for the group column
results_long$group <- factor(results_long$group, levels = c("high", "low"))

# Verify the levels
levels(results_long$group)
```

```{r}
# May need to winsorize the data

# Calculate the 95th percentile of the Abundance values
max_value <- quantile(results_long$Abundance, 0.95)

# Cap the values at the 95th percentile
results_long$Abundance <- ifelse(results_long$Abundance > max_value, max_value, results_long$Abundance)

# Remove rows with non-finite values in the Abundance column
results_long <- results_long[is.finite(results_long$Abundance), ]

# Calculate max value for setting label.y
max_value <- max(results_long$Abundance, na.rm = TRUE)

# Adjust p-value labels and position
results_long$group <- factor(results_long$group, levels = c("high", "low"))

# Plotting the data with adjusted p-value labels
p <- ggplot(results_long, aes(x = Cell_Type, y = Abundance, fill = group)) +
  geom_boxplot(outlier.size = 0.5) +
  stat_compare_means(aes(label = sprintf("p = %.3f", as.numeric(..p.format..))), method = "t.test", 
                     label.y = max_value - (max_value * 0.05), size = 2.5, hjust = 0.5) +
  labs(
    title = "Immune Cell Abundances - High & Low MKI67 Expression (MCP-counter)",
    y = "Cell Abundance",
    x = ""
  ) +
  scale_fill_manual(
    values = c("high" = "red", "low" = "turquoise")
  ) +
  scale_y_continuous(breaks = seq(0, max_value, max_value / 10), limits = c(0, max_value)) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    plot.title = element_text(hjust = 0.5),
    plot.margin = margin(10, 10, 10, 10)
  )

print(p)
```

```{r}
# Apply log transformation to the Abundance column
results_long$log_Abundance <- log1p(results_long$Abundance) # log1p handles log(0) by computing log(1 + x)

# Remove rows with non-finite values in the Abundance column
results_long <- results_long[is.finite(results_long$log_Abundance), ]

# Calculate max value for setting label.y
#max_value <- max(results_long$log_Abundance, na.rm = TRUE)

# Adjust p-value labels and position
results_long$group <- factor(results_long$group, levels = c("high", "low"))

# Plotting the data with adjusted p-value labels
p <- ggplot(results_long, aes(x = Cell_Type, y = log_Abundance, fill = group)) +
  geom_boxplot(outlier.size = 0.5) +
  stat_compare_means(aes(label = sprintf("p = %.3f", as.numeric(..p.format..))), method = "t.test", 
                     size = 2.5, hjust = 0.5) +
  labs(
    title = "Immune Cell Abundances - High & Low MKI67 Expression (MCP-counter)",
    y = "Log Cell Abundance",
    x = ""
  ) +
  scale_fill_manual(
    values = c("high" = "red", "low" = "turquoise")
  ) +
  scale_y_continuous(breaks = seq(0, max_value, max_value / 10), limits = c(0, max_value)) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    plot.title = element_text(hjust = 0.5),
    plot.margin = margin(10, 10, 10, 10)
  )

print(p)
```

Additional survival analyses with the aim of observing the impact of adjuvant mitotane therapy or other treatments on survival times

```{r}
# Filter the data for patients in quartiles 1 and 4
quartile_data <- finaldf %>%
  filter(quartile %in% c(1, 4))

# Convert Mitotane.Therapy to a factor
quartile_data$Mitotane.Therapy <- as.factor(quartile_data$Mitotane.Therapy)

# Separate data for quartile 1 and quartile 4
quartile1_data <- quartile_data %>%
  filter(quartile == 1)

quartile4_data <- quartile_data %>%
  filter(quartile == 4)

# Overall Survival analysis for quartile 1 based on Mitotane therapy
surv_model_q1 <- survfit(Surv(Overall.Survival..Months., OS) ~ Mitotane.Therapy, data = quartile1_data)

# Kaplan-Meier curve 1 for quartile 1
ggsurvplot(surv_model_q1, data = quartile1_data, pval = TRUE, palette = c("turquoise", "blue"),
           legend.labs = c("No Mitotane Therapy", "Mitotane Therapy"),
           title = "Overall Survival Curves for Quartile 1 based on Mitotane Therapy",
           xlab = "Time (months)", ylab = "Survival Probability",
           risk.table = TRUE, risk.table.col = "strata", tables.height = 0.3,
           theme = theme(plot.margin = margin(t = 2, r = 2, b = 2, l = 2, unit = "cm")),
           size = 0.8)

# Log-Rank test 1 for quartile 1
quartile1_logrank_test1 <- survdiff(Surv(Overall.Survival..Months., OS) ~ Mitotane.Therapy, data = quartile1_data)
quartile1_logrank_test1

# DSS analysis for quartile 1 based on Mitotane therapy
surv_model2_q1 <- survfit(Surv(Months.of.disease.specific.survival, DSS) ~ Mitotane.Therapy, data = quartile1_data)

# Kaplan-Meier curve 2 for quartile 1
ggsurvplot(surv_model2_q1, data = quartile1_data, pval = TRUE, palette = c("turquoise", "blue"),
           legend.labs = c("No Mitotane Therapy", "Mitotane Therapy"),
           title = "Disease Specific Survival Curves for Quartile 1 based on Mitotane Therapy",
           xlab = "Time (months)", ylab = "Survival Probability",
           risk.table = TRUE, risk.table.col = "strata", tables.height = 0.3,
           theme = theme(plot.margin = margin(t = 2, r = 2, b = 2, l = 2, unit = "cm")),
           size = 0.8)

# Log-Rank test 2 for quartile 1
quartile1_logrank_test2 <- survdiff(Surv(Months.of.disease.specific.survival, DSS) ~ Mitotane.Therapy, data = quartile1_data)
quartile1_logrank_test2
```


```{r}
# Overall Survival analysis for quartile 4 based on Mitotane therapy
surv_model1_q4 <- survfit(Surv(Overall.Survival..Months., OS) ~ Mitotane.Therapy, data = quartile4_data)

# Plot Kaplan-Meier curves for quartile 4
ggsurvplot(surv_model1_q4, data = quartile4_data, pval = TRUE, palette = c("red", "orange"),
           legend.labs = c("No Mitotane Therapy", "Mitotane Therapy"),
           title = "Overall Survival Curves for Quartile 4 based on Mitotane Therapy",
           xlab = "Time (months)", ylab = "Survival Probability",
           risk.table = TRUE, risk.table.col = "strata", tables.height = 0.3,
           theme = theme(plot.margin = margin(t = 2, r = 2, b = 2, l = 2, unit = "cm")),
           size = 0.8)

# Log-Rank test for quartile 4
quartile4_logrank_test1 <- survdiff(Surv(Overall.Survival..Months., OS) ~ Mitotane.Therapy, data = quartile4_data)
quartile4_logrank_test1

# DSS analysis for quartile 4 based on Mitotane therapy
surv_model2_q4 <- survfit(Surv(Months.of.disease.specific.survival, DSS) ~ Mitotane.Therapy, data = quartile4_data)

# Kaplan-Meier curve 2 for quartile 4
ggsurvplot(surv_model2_q4, data = quartile4_data, pval = TRUE, palette = c("red", "orange"),
           legend.labs = c("No Mitotane Therapy", "Mitotane Therapy"),
           title = "Disease Specific Survival Curves for Quartile 4 based on Mitotane Therapy",
           xlab = "Time (months)", ylab = "Survival Probability",
           risk.table = TRUE, risk.table.col = "strata", tables.height = 0.3,
           theme = theme(plot.margin = margin(t = 2, r = 2, b = 2, l = 2, unit = "cm")),
           size = 0.8)

# Log-Rank test 2 for quartile 4
quartile4_logrank_test2 <- survdiff(Surv(Months.of.disease.specific.survival, DSS) ~ Mitotane.Therapy, data = quartile4_data)
quartile4_logrank_test2
```
The lack of patients for each curve makes this a difficult challenge to prove.



